bigtable-hbase-parent/bigtable-hbase-integration-tests/src/test/java/com/google/cloud/bigtable/hbase/TestSnapshots.java

```diff
@@ -44,37 +44,51 @@
   final byte[] QUALIFIER = dataHelper.randomData("TestSingleColumnValueFilter");
 
   private final TableName tableName = IntegrationTests.newTestTableName();
-  private final String snapshotName = tableName.getNameAsString() + "_snp";
+  // The maximum size of a table id or snapshot id is 50. newTestTableName().size() can approach 50.
+  // Make sure that the snapshot name and cloned table are within the 50 character limit
+  private final String snapshotName = tableName.getNameAsString().substring(40) + "_snp";
   private final TableName clonedTableName =
-      TableName.valueOf(tableName.getNameAsString() + "_clone");
+      TableName.valueOf(tableName.getNameAsString().substring(40) + "_clone");
 
-   @After
-   public void delete() throws IOException {
-     Admin admin = getConnection().getAdmin();
-     admin.disableTable(tableName);
-     admin.deleteTable(tableName);
-     admin.disableTable(clonedTableName);
-     admin.deleteTable(clonedTableName);
-     admin.close();
-   }
+  @After
+  public void delete() throws IOException {
```

Comment:
nit: cleanup()? we already have a delete method: "delete(Admin admin, TableName tableName) "